test/server-side-test.js

```diff
@@ -1,81 +1,233 @@
-      table.boolean('is_banned').defaultTo(false);
-      table.boolean('is_active').defaultTo(true);
-      table.timestamp('date_created').defaultTo(knex.fn.now());
-      table.timestamp('date_updated').defaultTo(knex.fn.now());
-    }).then(() => {
-      console.log(('Created users table'));
-    });
+    knexdb.knex.schema.hasTable('users').then((exist) => {
+      if (!exist) {
+        knexdb.knex.schema.createTable('users', function(table) {
+          table.increments('id');
+          table.string('email').notNullable().unique('email').index();
+          table.string('first_name').notNullable();
+          table.string('last_name').notNullable();
+          table.string('phone_number').notNullable();
+          table.string('password_hash').notNullable();
+          table.boolean('is_admin').defaultTo(false);
+          table.boolean('is_banned').defaultTo(false);
+          table.boolean('is_active').defaultTo(true);
+          table.timestamp('date_created').defaultTo(knex.fn.now());
+          table.timestamp('date_updated').defaultTo(knex.fn.now());
+        }).then(() => {
+          console.log(('Created users table'));
+        }).catch((err) => {
+          console.log(err);
+        });
+      }
+    })
+  });
 
-    return (
-      new Users({
-        first_name: 'Duy',
-        last_name: 'Nguyen',
-        password_hash: 'hatch',
-        email: 'nguyenaiden@gmail.com',
-        phone_number: '+14086428264'
-      }).save() &&
-      
-      new Users({
-        first_name: 'Andi',
-        last_name: 'Oneto',
-        password_hash: 'hatch1',
-        email: 'andi@gmail.com',
-        phone_number: '+14158287474'
-      }).save()
-    )
+  after(function() {
+     return knexdb.knex.schema.dropTable('users');
   });
 
-  afterEach(function() {
+  describe('Data insertion', function() {
+    beforeEach(() => {
+      this.userSaveParams1 = {
+        first_name: 'John',
+        last_name: 'Doe',
+        password: 'hatch',
+        phone_number: '+14441114444',
+        email: 'John@gmail.com' 
+      };
 
-  });
+      this.userSaveParams2 = {
+        first_name: 'Jane',
+        last_name: 'Doe',
+        password: 'hatch1',
+        phone_number: '+14441114444',
+        email: 'Jane@gmail.com' 
+      }
+    })
 
-  after(function() {
-    return bookshelf.knex.schema.dropTable('users');
+    it('should be able to saveNewUser (test 1)', (done) => {
+      User.saveNewUser(this.userSaveParams1, usersModel)
+        .then((user) => {
+          expect(user.attributes.first_name).to.equal('John');
+          expect(user.attributes.last_name).to.equal('Doe');
+          done();
+        }, done);
+    })
+
+    it('should hash passwords upon entry (test 1)', (done) => {
+      User.getUserByEmail({ email: 'John@gmail.com'}, usersModel)
+        .then((user) => user.attributes.password_hash)
+          .then((passwordHash) => {
+            bcrypt.compare('hatch', passwordHash, (err, match) => {
```

Comment:
`bcrypt` has promises too! It can be cleaner:

```
 User.getUserByEmail({ email: 'John@gmail.com'}, usersModel)
    .then(user => user.attributes.password_hash)
    .then(passwordHash => bcrypt.compare('hatch',  passwordHash))
    .then(match => {
         expect(match).to.be.true;
         done();
     });
```